server/schedulers/balance_leader.go

```diff
@@ -137,6 +141,101 @@ func (l *balanceLeaderScheduler) IsScheduleAllowed(cluster schedule.Cluster) boo
 	return allowed
 }
 
+type batchController struct {
+	cluster     schedule.Cluster
+	sources     []*core.StoreInfo
+	targets     []*core.StoreInfo
+	sourceIndex int
+	targetIndex int
+	plan        *balancePlan
+	usedRegions map[uint64]struct{}
+}
+
+func (bc *batchController) initSort() {
+	sort.Slice(bc.sources, func(i, j int) bool {
+		iOp := bc.plan.GetOpInfluence(bc.sources[i].GetID())
+		jOp := bc.plan.GetOpInfluence(bc.sources[j].GetID())
+		return bc.sources[i].LeaderScore(bc.plan.kind.Policy, iOp) >
+			bc.sources[j].LeaderScore(bc.plan.kind.Policy, jOp)
+	})
+	sort.Slice(bc.targets, func(i, j int) bool {
+		iOp := bc.plan.GetOpInfluence(bc.targets[i].GetID())
+		jOp := bc.plan.GetOpInfluence(bc.targets[j].GetID())
+		return bc.targets[i].LeaderScore(bc.plan.kind.Policy, iOp) <
```

Comment:
How about defining two functions? With the name of the function, the maintainer can easily find the slice is ordered by desc or asc.